### DISCUSSION TITLE
IRL App Connecting

### DISCUSSION THREAD

**Discussion by ajbott** [PullDating/Org#199] [Ideas & Features]
## Overview

The IRL App Connecting feature introduces a QR code functionality for use with in-person meeting situations, similar to e.g. the Venmo function to scan and send money. It would allow people who randomly met *first* in real life to then immediately match on the dating app as a quick, safe, and easy way to connect and continue their interaction. 

This feature aims to enhance safety, streamline connections, and bridge the gap between online and in-person interactions.

## User Journey
- A user is in an in-person setting
- They meet someone they are interested in
- They introduce the app and show their unique QR code to connect with the other person instantly
- The other user does not have to have the app, if they don't it prompts them to download it

### QR Code for Instant Connection
- Users can generate a QR code within the app.
- The QR code contains information about the user's interests and preferences.
- The code is saved to the user's device for quick access.

### Safer Connections
- Users can share their QR code instead of personal phone numbers.
- Instantly adds the person to the user's matches on the app.

### Community Health Check
- Checking if a user has been banned before provides insights into their community behavior.
- Promotes a healthier dating environment.

### Streamlined Connections
- Simplifies the process of adding someone to matches during in-person meetings.

### Enhanced Safety
- Reduces the need to share personal contact information initially.

### Community Awareness
- Users can make informed decisions based on the other person's community history.

## Usage

- Users can share their QR code by displaying it on their device for others to scan.
- Scanning the QR code instantly adds the person to matches on the app.

## Conclusion

The IRL App Connecting feature offers a convenient and safe way for users to connect in person, fostering streamlined interactions and promoting community awareness.

**Comment by WilliamDormer** [comment]
I really like this idea. A quicker replacement for asking for the phone number. They just save the QR code to their phone, or have it on the main page when they open the app button.

**Reply by ajbott** [reply]
Additionally, it offers enhanced safety. Instead of exchanging personal phone numbers, users can communicate securely through the app, maintaining their privacy. Moreover, users have the option to conceal their profile information, allowing them to control the information visible to others.

Incorporating this feature on the main page is a great concept. Furthermore, implementing a widget would enhance user convenience. By simply tapping the widget, users can instantly access their contact ID. This could potentially lead to the evolution of the app into a social communication platform, akin to the successful model of VRChat's friends groups.

**Comment by Oshyan** [comment]
This is a very interesting idea! I think it's unlikely to be used much early on for that reason might be a lower priority for implementation. But at the same time it's an interesting growth function so if it's relatively easy to implement, then it's kind of brilliant in that respect. I'm a little uncertain how incentivized people would be to use it, but if it's designed and pitched/framed well in the app and other messaging, it has real potential. Well done!

**Comment by Oshyan** [comment]
I edited the main description here to make it clearer how this works, because I didn't understand it was about *new* IRL connections at first.

I also have some concerns about a couple specific features/points made above:
>The QR code contains information about the user's interests and preferences.

What does this mean, why does it work this way? Isn't the QR code just basically a link to their profile in the app that *already* has "info about the user's interests and preferences"? If that's the intent then just stating it as a link to their profile is probably clearer.

>Checking if a user has been banned before provides insights into their community behavior.

How exactly do you imagine the actual experience of this working? The way I am imagining the situation, you have two people who have just met for the first time, with no prior dating app interaction. They go to match with this QR code, and then one person sees that the other person has a Flag or Warning or whatever on their profile. What do they do next in that immediate, in-person interaction with the person standing right in front of them? Isn't that awkward if not dangerous? 

I don't really think that it should even be a specific feature of this QR code function in fact, because user profile flags/warnings/whatever will almost certainly already be a component of the app. So in that respect this potential awkwardness is unavoidable: if they match in-person, in that moment, and they happen to actually look at the other person's profile, even a glance, they will probably see stuff like flags/warnings. In some sense that is a good thing, it might save them from a potentially dangerous real-world situation, i.e. they could with that information choose to extricate themselves from the situation. But it *could* also be fairly awkward, if not dangerous. I am a bit torn on it, really... But regardless I don't think that described feature is really a component of *this* feature, rather an outcome of the overall app feature set and the way in which this feature works and interacts with other features.
